src/c2s/mongoose_c2s.erl

```diff
@@ -247,12 +264,30 @@ handle_socket_packet(StateData = #c2s_data{parser = Parser}, Packet) ->
 -spec handle_socket_elements(data(), [exml:element()], non_neg_integer()) -> fsm_res().
 handle_socket_elements(StateData = #c2s_data{shaper = Shaper}, Elements, Size) ->
     {NewShaper, Pause} = mongoose_shaper:update(Shaper, Size),
-    mongoose_metrics:update(global, [data, xmpp, received, xml_stanza_size], Size),
+    [mongoose_instrument:execute(xmpp_stanza_size_received, #{},
+                                 #{byte_size => elem_size(El)})
+     || El <- Elements],
     NewStateData = StateData#c2s_data{shaper = NewShaper},
     MaybePauseTimeout = maybe_pause(NewStateData, Pause),
     StreamEvents = [ {next_event, internal, XmlEl} || XmlEl <- Elements ],
     {keep_state, NewStateData, MaybePauseTimeout ++ StreamEvents}.
 
+elem_size(#xmlstreamerror{name = Name}) ->
+    byte_size(Name);
+elem_size(El) ->
+    try
+        exml:xml_size(patch_element(El))
+    catch _Class:Reason:Stacktrace ->
+        ?LOG_ERROR(#{what => failed_to_xml_size, element => El}),
+        0
```

Comment:
Just don't put `byte_size` in the measurements instead of the zero.

Reply:
this `elem_size` function is funny.
Basically, I added it because exml:xml_size/1 is broken for incoming stanzas elemnt, that is why we have patch_element. Need to fix exml:xml_size/1 in a separate PR to exml.

This crash:
```
when=2024-06-17T20:33:07.738636+00:00 level=error pid=<0.2437.0> at=supervisor:do_restart/3:757 report="[{supervisor,{local,mongoose_c2s_sup}},{errorContext,child_terminated},{reason,{{{xmlstreamstart,<<\"stream:stream\">>,
 [{<<\"from\">>,undefined},{<<\"to\">>,<<\"localhost\">>},{<<\"version\">>,<<\"1.0\">>},{<<\"xml:lang\">>,<<\"en\">>},{<<\"xmlns\">>,<<\"jabber:client\">>},{<<\"xmlns:stream\">>,<<\"http://etherx.jabber.org/streams\">>}]},
badarg,[{erlang,byte_size,[undefined],
[{error_info,#{module => erl_erts_errors}}]},{exml,xml_size,1,[{file,\"/Users/mikhailuvarov/erlang/esl/MongooseIM/_build/default/lib/exml/src/exml.erl\"},{line,56}]},{exml,xml_size,1,[{file,\"/Users/mikhailuvarov/erlang/esl/MongooseIM/_build/default/lib/exml/src/exml.erl\"},{line,39}]},{exml,xml_size,1,[{file,\"/Users/mikhailuvarov/erlang/esl/MongooseIM/_build/default/lib/exml/src/exml.erl\"},{line,50}]},{mongoose_c2s,'-handle_socket_elements/3-lc$^0/1-0-',1,[{file,\"/Users/mikhailuvarov/erlang/esl/MongooseIM/src/c2s/mongoose_c2s.erl\"},{line,251}]},{mongoose_c2s,handle_socket_elements,3,[{file,\"/Users/mikhailuvarov/erlang/esl/MongooseIM/src/c2s/mongoose_c2s.erl\"},{line,255}]},{gen_statem,loop_state_callback,11,[{file,\"gen_statem.erl\"},{line,1379}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,241}]}]},[{mongoose_c2s,'-handle_socket_elements/3-lc$^0/1-0-',1,[{file,\"/Users/mikhailuvarov/erlang/esl/MongooseIM/src/c2s/mongoose_c2s.erl\"},{line,253}]},{mongoose_c2s,handle_socket_elements,3,[{file,\"/Users/mikhailuvarov/erlang/esl/MongooseIM/src/c2s/mongoose_c2s.erl\"},{line,255}]},{gen_statem,loop_state_callback,11,[{file,\"gen_statem.erl\"},{line,1379}]},{proc_lib,init_p_do_apply,3,[{file,\"proc_lib.erl\"},{line,241}]}]}},{offender,[{pid,<0.4003.0>},{id,undefined},{mfargs,{mongoose_c2s,start_link,undefined}},{restart_type,temporary},{significant,false},{shutdown,brutal_kill},{child_type,worker}]}]" label={supervisor,child_terminated}
```

Reply:
Also,
```
-spec handle_socket_elements(data(), [exml:element()], non_neg_integer()) -> fsm_res().
```
This spec for second argument is wrong, there is xmlstreamstart, end and error.